lib/oli_web/components/delivery/actions/actions.ex

```diff
@@ -0,0 +1,69 @@
+defmodule OliWeb.Components.Delivery.Actions do
+  use Surface.LiveComponent
+
+  alias Lti_1p3.Tool.ContextRoles
+  alias Oli.Accounts
+
+  prop enrollment_info, :map, required: true
+  prop section_slug, :string, required: true
+  prop user, :map, required: true
+
+  data enrollment, :map, default: %{}
+  data is_instructor, :boolean, default: false
+
+  def update(
+        %{user: user, section_slug: section_slug, enrollment_info: enrollment_info} = _assigns,
+        socket
+      ) do
+    {:ok,
+     assign(socket,
+       enrollment: enrollment_info.enrollment,
+       is_instructor: enrollment_info.is_instructor,
+       section_slug: section_slug,
+       user: user
+     )}
+  end
+
+  def render(assigns) do
+    ~F"""
+      <div class="mx-10 mb-10 bg-white shadow-sm">
+
+        <div class="flex flex-col sm:flex-row sm:items-end px-6 py-4 border instructor_dashboard_table">
+          <h4 class="pl-9 !py-2 torus-h4 mr-auto dark:!text-black">Actions</h4>
+        </div>
+
+        <div class="flex justify-between items-center px-14 py-8">
+          <div class="flex flex-col">
+            <span class="dark:text-black">Change role to Instructor</span>
+            <span class="text-xs text-gray-400 dark:text-gray-950">If this option is enabled, the user will have the role of Instructor in this course section.</span>
+          </div>
+          <label class="relative inline-flex items-center cursor-pointer">
+            <input
```

Comment:
Instead of a checkbox, please make this a HTML SELECT with two options "Instructor" and "Student".  When the user clicks to change, we must prompt them with a dialog box asking to confirm.

The reason to move towards an HTML SELECT is that very shortly we will be adding additional roles that an enrolled user can have.

Reply:
Sure, I'm fine thinking about possible future scenarios!